### PR TITLE
Added ShopChest Protection Module

### PR DESCRIPTION
**Pull request by EpicPlayerA10**
## Description
Added ShopChest Protection Module

### CODE REVIEW

File: dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/ShopChestProtectionModule.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Override
+    public boolean hasPermission(OfflinePlayer p, Location l, Interaction action) {
+        switch (action) {
+            case BREAK_BLOCK:
+                return !shopChest.getShopUtils().isShop(l);
+            default:
+                return true;
+        }
```
Comment (md5sha256) [inline dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/ShopChestProtectionModule.java:39]: If you only have two cases, it doesn't make sense to use a switch. Consider using an if else instead
```suggestion
    public boolean hasPermission(OfflinePlayer p, Location l, Interaction action) {
        if (action == Interaction.BREAK_BLOCK) {
                return !shopChest.getShopUtils().isShop(l);
        } else {
                return true;
        }
```

**Review by md5sha256 — CHANGES_REQUESTED** [review]

**Comment by md5sha256** [comment]
LGTM

File: dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/ShopChestProtectionModule.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import io.github.bakedlibs.dough.protection.Interaction;
+import io.github.bakedlibs.dough.protection.ProtectionModule;
+
+import org.bukkit.Location;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.plugin.Plugin;
+
+import javax.annotation.Nonnull;
+
+public class ShopChestProtectionModule implements ProtectionModule {
```
Comment (TheBusyBiscuit) [inline dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/ShopChestProtectionModule.java:14]: Missing author tag and javadocs

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Looks good implementation-wise.
We just need some documentation before I will merge this

**Review by J3fftw1 — APPROVED** [review]
LGTM

**Review by J3fftw1 — DISMISSED** [review]
LGTM

**Review by Sfiguz7 — DISMISSED** [review]

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**